### ISSUE TITLE
Firebase offline persistence

### ISSUE THREAD

**Issue by KMCGamer** [SCCapstone/Cupboard#62]
Firebase can be used offline indefinitely so we may want to remove all UUIDs and just use firebase ids instead.

[![](https://fireassets.firebaseapp.com/resources/images/blog/offline.gif)](https://firebase.googleblog.com/2015/05/announcing-mobile-offline-support_93.html)

As the gif shows, firebase can store information locally, offline, forever if you want it to, and then when the phone goes online, it can sync automatically.

Basically, to simplify everything we should probably remove UUIDs and just use firebase ids.

https://firebase.googleblog.com/2015/05/announcing-mobile-offline-support_93.html
https://firebase.google.com/docs/database/android/offline-capabilities

**Comment by KMCGamer** [comment]
Actually, i might have to do more research on this. I'm finding conflicting advice on people managing their own local storage or just letting firebase do all of it.

**Comment by KMCGamer** [comment]
This is being implemented in local-db branch. Decided to go with Room (SQLite) local database alongside RXJava2 for reactive programming. Local database and Firebase database will sync with each other automatically through the Observable/Observer pattern with the help of RXJava.

**Comment by WillWells** [comment]
Fully implemented as of Release 1.0
